src/vs/workbench/contrib/externalUriOpener/common/externalUriOpenerService.ts

```diff
@@ -110,22 +113,44 @@ export class ExternalUriOpenerService extends Disposable implements IExternalUri
 			}
 		}));
 		if (validOpeners.length === 0) {
-			return false;
+			return [];
 		}
 
 		// See if we have a preferred opener first
 		const preferred = firstOrDefault(validOpeners.filter(x => x.priority === modes.ExternalUriOpenerPriority.Preferred));
 		if (preferred) {
-			return preferred.opener.openExternalUri(targetUri, ctx, token);
+			return [preferred.opener];
 		}
 
 		// See if we only have optional openers, use the default opener
-		if (validOpeners.every(x => x.priority === modes.ExternalUriOpenerPriority.Option)) {
+		if (!allowOptional && validOpeners.every(x => x.priority === modes.ExternalUriOpenerPriority.Option)) {
```

Comment:
The change I made here is around picking _any_ non-default opener. I wanted to have a way to say "definitely open this not in the browser". Of course, I still want all your good logic around picking the best opener, but if there is no best opener, then I just want one of them.